### PR TITLE
11.0 backport account document reversal

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Backport to 11.0 of https://github.com/OCA/account-financial-tools/pull/820

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
FYI @kittiu

File: account_document_reversal/models/account_bank_statement.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                raise UserError(
+                    _('Only new bank statement can be cancelled'))
+            return self.reverse_document_wizard()
+        return super().button_cancel_reconciliation()
+
+    @api.multi
+    def action_document_reversal(self, date=None, journal_id=None):
+        """ Reverse all moves related to this statement + delete payment """
+        # This part is from button_cancel_reconciliation()
+        aml_to_unbind = self.env['account.move.line']
```
Comment (nikul-serpentcs) [inline account_document_reversal/models/account_bank_statement.py:28]: @jbeficent  here both way are working same but Duplicate Variable Declaration
So, trying to ` aml_to_unbind = aml_to_cancel = self.env['account.move.line']`

Same as a for `self.env['account.payment']`

**Review by nikul-serpentcs — COMMENTED** [review]
nitpicking

**Review by etobella — APPROVED** [review]
Great addition. Tested and It works like a charm :+1:

**Comment by JordiBForgeFlow** [comment]
@kittiu can you have a look into it? just to make sure that I have not left anything out.

**Comment by kittiu** [comment]
Looks good. +1

**Pull request merged**